core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestSampler.kt

```diff
@@ -190,18 +201,176 @@ class RestSampler : Sampler<RestIndividual>() {
+                            .map{ it.name }
+                            .toSet().containsAll((g as ObjectGene).fields.map { it.name }) }.toMutableMap()
+                if (restrictedModels.isEmpty()) return Pair(ObjectGene("none", mutableListOf()), Pair("none", "Not_found"))
+                val ret = randomness.choose(restrictedModels)
+                return Pair(ret, Pair(ret.name, "Complete_object"))
+            }
+            else -> {
+                modelCluster.forEach { k, model ->
+                    val fields = model.fields.filter { field ->
+                        when (field::class) {
+                            OptionalGene::class -> g::class === (field as OptionalGene).gene::class
+                            else -> g::class === field::class
+                        }
+                    }
+                    restrictedModels[k] = ObjectGene(model.name, fields)
+                }
+            }
+        }
 
-        return RestIndividual(actions, SampleType.RANDOM)
+        //Having filtered the objects, build the probability map.
+        val likely = likelyhoodsExtended(g.getVariableName(), restrictedModels)
+                .toList()
+                .sortedBy { (_, value) -> -value}
+                .toMap()
+
+        if (likely.isNotEmpty()){
+            // there is at least one likely match
+            val selected = pickObj((likely as MutableMap<Pair<String, String>, Float>), probabilistic = true)
+            val selObject = modelCluster.get(selected.first)!!
+            return Pair(selObject, selected)
+        }
+        else{
+            // there is no likely match
+            val fields = listOf(g)
+            val wrapper = ObjectGene("Gene_wrapper_object", fields)
+            return Pair(wrapper, Pair("", "Single_gene"))
+        }
     }
 
+    private fun randomizeActionGenes(action: Action, probabilistic: Boolean = false) {
+        if(!config.enableCompleteObjects) {
+            action.seeGenes().forEach { it.randomize(randomness, false) }
+        }
+        else {
+            action.seeGenes().forEach { g ->
+                /*Obtain the object proposed for mutation. Can be:
+                1. Complete object - the entire object is used and needs to be mutated.
+                2. Just the gene g (wrapped in an ObjectGene) - an object match could not be found, g is mutated as such.
+                3. The object, plus a 2 string pair - model name, gene name. This identifies an object that needs to be mutated
+                and which of its genes will be used.
+                */
+
+                val innerGene = when (g::class){
+                    OptionalGene::class -> (g as OptionalGene).gene
+                    DisruptiveGene::class -> (g as DisruptiveGene<*>).gene
+                    else -> g
+                }
+
+                val (proposed, field) = proposeObject(innerGene)
+
+                when(field.second) {
+                    "Single_gene" -> {
```

Comment:
we should try to avoid "magic" strings... where is that one coming from? likely could put it in a shared final variable